src/blocks/VoiceApp.tsx

```diff
@@ -0,0 +1,335 @@
+
+      // Get media recorder
+      const currentMediaRecorder = mediaRecorder.current
+
+      // Get the stream
+      const stream = currentMediaRecorder.stream;
+
+      // Create a new context
+      let context = new AudioContext();
+
+      // Create a new node
+      let node = context.createMediaStreamSource(stream);
+
+
+      start({ node, context, getPlayable: () => currentMediaRecorder.state === "recording" })
+    }
+  }
+
+  const handleSubmit = () => {
+    onSave(audio)
+  }
+
+  return (
+    <Box style={{
+      backgroundColor: 'none', padding: '20px', display: 'flex', justifyContent: 'center', alignItems: 'flex-end', gap: '20px', width: '100%'
+    }} >
+      <Fab onClick={toggleRecord} sx={{ width: { xs: "150px", md: "250px", lg: "250px" }, height: { xs: "150px", md: "250px", lg: "250px" } }}
+        style={
+          isRecording ?
+            { backgroundColor: 'white', border: "5px solid black" } :
+            { backgroundColor: "black" }
+        }>
+        <MicIcon style={isRecording ? { color: 'black', } : { color: 'white', }} />
+      </Fab>
+      <Fab sx={{
+        width: { xs: "75px", md: "150px", lg: "150px" }, height: { xs: "75px", md: "150px", lg: "150px" }, color: 'white', padding: '20px'
+      }} style={{
+        backgroundColor: 'red'
+      }} onClick={() => {
+        audio === "" ? alert("Record some audio first") : handleSubmit();
+      }}>
+        Preview
+      </Fab>
+
+    </Box>
+  )
+}
+
+const AudioCard = () => {
+
+  return (
+    <Card style={{ backgroundColor: 'black', borderRadius: '30px', width: "100%", display: 'flex', flexDirection: 'column', justifyContent: 'center', alignItems: 'center' }} sx={{ height: { xs: "350px", md: "400px", lg: "400px" } }} >
+      <CardContent style={{ backgroundColor: 'black', padding: '24px', height: "100%", width: "100%" }}>
+        <Box style={{
+          color: 'black', backgroundColor: 'none', display: 'flex', justifyContent: 'space-between', width: '100%', height: "100%", alignItems: 'center'
+        }}>
+          <canvas style={{ color: 'white', width: "100%", height: "100%" }} id="oscilloscope"></canvas>
+        </Box>
+      </CardContent>
+    </Card>
+  )
+}
+
+/* 
+
+  SEAM CLASS
+
+*/
+
+export default class VoiceBlock extends Block {
+
+  render() {
+    return <PostInFeed url={this.model.data["audio"]} />;
+  }
+
+  renderEditModal(done: (data: BlockModel) => void) {
+    const handleSave = (audio: string) => {
+      this.model.data["audio"] = audio
+      done(this.model)
+    }
+
+    return (
+      <AudioCtx>
+        <div style={{ maxWidth: "100vw", overflow: "visible", display: "flex", flexDirection: "column", height: "100%", alignItems: "center", justifyContent: "space-around" }}>
+          <AudioCard />
+          <AudioButtons onSave={handleSave} />
+        </div>
+      </AudioCtx>
+    )
+  }
+
+  renderErrorState() {
+    // Shouldn't have to use this anywhere because all types should be properly narrowed
```

Comment:
can use this function to support the above comments around not springing alerts to the user

Reply:
understood, fixing now